From 0e2d07bdd3bc9daed1002ed4a2a6f8e0cfb7e808 Mon Sep 17 00:00:00 2001
From: frq09432 <nicolas.louboutin@st.com>
Date: Thu, 16 Jan 2020 08:43:43 +0100
Subject: [PATCH 2/4] drm_hwcomposer: Insure that the layer is compatible with
 the plan capability

  By default, consider that the plan is not compatible
  TODO: manage the compatibility dynamically

Change-Id: I864e213b4cdfaf2b91dacb00f94fd7671dd80672
---
 drmhwctwo.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drmhwctwo.cpp b/drmhwctwo.cpp
index 95370d5..1fb6ef1 100644
--- a/drmhwctwo.cpp
+++ b/drmhwctwo.cpp
@@ -800,6 +800,12 @@ HWC2::Error DrmHwcTwo::HwcDisplay::ValidateDisplay(uint32_t *num_types,
   if (avail_planes < layers_.size())
     avail_planes--;
 
+/* By default, if only one layer available, consider that it has not a compatible format
+   * TODO: Check that the validated layer has a compatible format
+   */
+  if (layers_.size() == 1)
+    avail_planes--;
+
   std::map<uint32_t, DrmHwcTwo::HwcLayer *, std::greater<int>> z_map;
   for (std::pair<const hwc2_layer_t, DrmHwcTwo::HwcLayer> &l : layers_)
     z_map.emplace(std::make_pair(l.second.z_order(), &l.second));
-- 
2.7.4

